lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/ar/AccessRestrictionStatusDaoImplTest.java

```diff
@@ -149,16 +157,36 @@ public void testGetEntityStatusWithNullUserId() {
 		Long userId = null;
 		String message = assertThrows(IllegalArgumentException.class, () -> {
 			// call under test
-			accessRestrictionStatusDao.getEntityStatus(subjectIds, userId);
+			accessRestrictionStatusDao.getEntityStatus(subjectIds, userId, Collections.emptySet());
 		}).getMessage();
 		assertEquals("userId is required.", message);
 	}
```

Comment:
needed tests:
Case where there are multiple entities that share one ar has eligible and one ar not eligible.
Case where a user has been granted eligible through multiple principals.
Case where multiple objects of different types have the same id have acls.
Case where the ALC grant non-exempt permission to the same user.
Case where there is an exemption on the AR but not for that user.